packages/jest-reporters/src/SummaryReporter.ts

```diff
@@ -54,13 +52,28 @@ const {npm_config_user_agent, npm_lifecycle_event, npm_lifecycle_script} =
 export default class SummaryReporter extends BaseReporter {
   private _estimatedTime: number;
   private readonly _globalConfig: Config.GlobalConfig;
+  private readonly _summaryThreshold: number;
 
   static readonly filename = __filename;
 
-  constructor(globalConfig: Config.GlobalConfig) {
+  constructor(
+    globalConfig: Config.GlobalConfig,
+    options: SummaryReporterOptions = {},
+  ) {
     super();
     this._globalConfig = globalConfig;
     this._estimatedTime = 0;
+    this.validateOptions(options);
+    this._summaryThreshold = options.summaryThreshold || 20;
+  }
+
+  private validateOptions(options: SummaryReporterOptions) {
+    if (
+      options.summaryThreshold &&
```

Comment:
Would be good to prepend underscore just like in other private methods. Optional chaining and nullish coalescing also looks good here.

```suggestion
    options?: SummaryReporterOptions,
  ) {
    super();
    this._globalConfig = globalConfig;
    this._estimatedTime = 0;
    this._validateOptions(options);
    this._summaryThreshold = options?.summaryThreshold ?? 20;
  }

  private _validateOptions(options?: SummaryReporterOptions) {
    if (
      options?.summaryThreshold &&
```